olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp

```diff
@@ -40,7 +40,7 @@ std::string CreateKey(const std::string& hrn, const std::string& layer_id,
                       const std::string& partitionId,
                       const boost::optional<int64_t>& version) {
   return hrn + "::" + layer_id + "::" + partitionId +
-         "::" + (version ? std::to_string(*version) + "::partition" : "");
+         "::" + (version ? std::to_string(*version) + "::" : "") + "partition";
 }
 std::string CreateKey(const std::string& hrn, const std::string& layer_id,
```

Comment:
Same here. The layer metadata should also include `partitions` at the end.

Reply:
added partitions and fixed Clear function.